lib/service/service.go

```diff
@@ -679,7 +679,15 @@ func NewTeleport(cfg *Config) (*TeleportProcess, error) {
 		} else {
 			switch cfg.JoinMethod {
 			case types.JoinMethodToken, types.JoinMethodUnspecified, types.JoinMethodIAM:
-				cfg.HostUUID = uuid.NewString()
+				// Checking error instead of the usual uuid.New() in case uuid generation
+				// fails due to not enough randomness.
+				rawID, err := uuid.NewRandom()
+				if err != nil {
+					return nil, trace.Errorf("Teleport failed to generate host UUID." +
+						" This may happen if randomness source is not fully initialized " +
+						"when the node is starting up. Please try restarting Teleport again")
```

Comment:
Suggestion:

```suggestion
					return nil, trace.Errorf("" +
						"Teleport failed to generate host UUID. " +
						"This may happen if randomness source is not fully initialized when the node is starting up. " +
						"Please try restarting Teleport again.")
```